### PR TITLE
feat!: move API to python microgenerator

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

**Comment by xiaozhenliu-gg5** [comment]
Kokoro fails only for session cover
```
Running session cover
Creating virtual environment (virtualenv) using python3.8 in .nox/cover
pip install coverage pytest-cov
coverage report --show-missing --fail-under=100
No data to report.
Command coverage report --show-missing --fail-under=100 failed with exit code 1
Session cover failed.
```
Can I get some advice here? Thanks! @arithmetic1728 @busunkim96

File: UPGRADING.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+# TODO: Initialize `dashboard`:
+dashboard = {}
+response = client.create_dashboard(parent, dashboard)
+```
+
+
+**After:**
+```py
+from google.cloud.monitoring_dashboard import v1
+client = v1.DashboardsServiceClient()
```
Comment (arithmetic1728) [inline UPGRADING.md:44]: There is a namespace change, so the `after` code would be:
```
from google.cloud import monitoring_dashboard_v1
client = monitoring_dashboard_v1.DashboardsServiceClient()
```

I would recommend add a new section above explaining there is a namespace change, and give the `before` and `after` code explaining how to create a client. In this section, only keep the `response = client.create_dashboard(...)` part. The script itself only modifies the method code, it doesn't fix the namespace.

File: UPGRADING.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+This call is invalid because it mixes `request` with a keyword argument `dashboard`. Executing this code will result in an error.
+
+```py
+response = client.create_dashboard(
+    request={
+        "parent": parent,
+    },
+    dashboard=dashboard
+)
+```
```
Comment (arithmetic1728) [inline UPGRADING.md:110]: I remember there is a section after this about `Enums and Types`.

File: setup.py [hunk 3]
```diff
@@ -25,7 +25,9 @@
 version = "1.0.0"
 release_status = "Development Status :: 5 - Production/Stable"
 dependencies = [
-    "google-api-core[grpc] >= 1.14.0, < 2.0.0dev",
+    "google-api-core[grpc] >= 1.22.0, < 2.0.0dev",
+    "proto-plus >= 0.4.0",
+    "libcst >= 0.2.5",
     'enum34; python_version < "3.4"',
```
Comment (arithmetic1728) [inline setup.py:31]: 'enum34; python_version < 3.4' is no longer needed since python is not required to >= 3.6

File: setup.py [hunk 4]
```diff
@@ -58,7 +62,6 @@
         "License :: OSI Approved :: Apache Software License",
         "Programming Language :: Python",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.5",
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
```
Comment (arithmetic1728) [inline setup.py:66]: add "Programming Language :: Python :: 3.8"?

File: setup.py [hunk 5]
```diff
@@ -68,7 +71,8 @@
     packages=packages,
     namespace_packages=namespaces,
     install_requires=dependencies,
-    python_requires=">=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*",
+    python_requires=">=3.6,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*",
```
Comment (arithmetic1728) [inline setup.py:74]:
```suggestion
    python_requires=">=3.6",
```

Comment (xiaozhenliu-gg5) [reply setup.py]: done.

File: setup.py (hunk 4 above)
Comment (xiaozhenliu-gg5) [reply setup.py:65]: done, thanks!

File: setup.py (hunk 3 above)
Comment (xiaozhenliu-gg5) [reply setup.py]: sure, removed.

File: UPGRADING.md (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply UPGRADING.md]: Sounds reasonable to me, I removed the creating client part here and added a section above to explain the breaking change in the sense of creating new client. Please LMK if it looks good to you, thanks.

File: UPGRADING.md (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply UPGRADING.md:119]: Yes, while I searched `enums` in code base, and seems we don't use enums anywhere, adding it to the upgrading doc might confuse users, WDYT? Thanks!

Comment (busunkim96) [reply UPGRADING.md:121]: Given that the present directory layout is borked I think you're right that it'll be more confusing to have that section.

File: UPGRADING.md (hunk 1 above)
Comment (arithmetic1728) [reply UPGRADING.md]: It looks good to me, thanks!

File: UPGRADING.md (hunk 2 above)
Comment (arithmetic1728) [reply UPGRADING.md:121]: yes that makes sense.

**Pull request merged**